src/pfe/portal/routes/projects/metrics.route.js

```diff
@@ -197,6 +204,21 @@ router.post('/api/v1/projects/:id/metrics/inject', validateReq, async function (
   }
 });
 
+async function copyProjectToBuild(project){
+  const globalProjectPath = path.join(project.workspace, project.name);
+  const projectRoot = cwUtils.getProjectSourceRoot(project);
+  if (project.buildStatus != "inProgress") {
+    await cwUtils.copyProjectContents(
+      project,
+      globalProjectPath,
+      projectRoot
+    );
+  } else {
+    // if a build is in progress, wait 5 seconds and try again
+    await cwUtils.timeout(5000)
```

Comment:
Not sure what to do about this but if a long build (liberty) is in progress when metrics injection is turned on then `/api/v1/projects/:id/metrics/inject` could take a really long time to return.